pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -74,16 +76,98 @@ type cloudWatchExecutor struct {
 
 func (e *cloudWatchExecutor) newSession(region string) (*session.Session, error) {
 	dsInfo := e.getDSInfo(region)
-	creds, err := getCredentials(dsInfo)
+
+	cacheKey := fmt.Sprintf("%d:%s:%s:%s:%s:%s", dsInfo.AuthType, dsInfo.AccessKey, dsInfo.Profile, dsInfo.AssumeRoleARN,
+		dsInfo.ExternalID, region)
+
+	sessCacheLock.RLock()
+	if env, ok := sessCache[cacheKey]; ok {
+		if env.expiration.After(time.Now().UTC()) {
+			sessCacheLock.RUnlock()
+			plog.Debug("Using cached AWS session", "key", cacheKey)
+			return env.session, nil
+		}
+	}
+	sessCacheLock.RUnlock()
+
+	cfgs := []*aws.Config{
+		{
+			CredentialsChainVerboseErrors: aws.Bool(true),
+		},
+	}
+
+	var regionCfg *aws.Config
+	if dsInfo.Region == defaultRegion {
+		plog.Warn("Region is set to \"default\", which is unsupported")
+		dsInfo.Region = ""
+	}
+	if dsInfo.Region != "" {
+		regionCfg = &aws.Config{Region: aws.String(dsInfo.Region)}
+		cfgs = append(cfgs, regionCfg)
+	}
+
+	switch dsInfo.AuthType {
+	case authTypeSharedCreds:
+		plog.Debug("Authenticating towards AWS with shared credentials", "profile", dsInfo.Profile,
+			"region", dsInfo.Region)
+		cfgs = append(cfgs, &aws.Config{
+			Credentials: credentials.NewSharedCredentials("", dsInfo.Profile),
+		})
+	case authTypeKeys:
+		plog.Debug("Authenticating towards AWS with an access key pair", "region", dsInfo.Region)
```

Comment:
These log statements that mentions AWS may not be needed since you have the logger name of `tsdb.cloudwatch`? So could be changed to "Authenticating with an access key pair"?

Reply:
@marefr I prefer being explicit here, it helps disambiguate when reading the logs, so don't see the problem. Quite the opposite, with your change I might be unsure when reading the logs exactly what's being authenticated against since you don't have the context of the code. @papagian WDYT?

Reply:
Sure. No biggie really